src/Catalyst.Abstractions/Ledger/IWeb3EthApi.cs

```diff
@@ -32,5 +37,26 @@ public interface IWeb3EthApi
         IStateReader StateReader { get; }
         IDeltaResolver DeltaResolver { get; }
         IStateRootResolver StateRootResolver { get; }
+        IDeltaCache DeltaCache { get; }
+
+        object SyncRoot { get; }
+        ITransactionProcessor Processor { get; }
+        IStorageProvider StorageProvider { get; }
+        IStateProvider StateProvider { get; }
+    }
+
+    public static class Web3EthApiExtensions
+    {
+        public static Delta GetLatestDelta(this IWeb3EthApi api)
+        {
+            // change to appropriate hash
+            Cid cid = api.DeltaResolver.LatestDelta;
+            if (!api.DeltaCache.TryGetOrAddConfirmedDelta(cid, out Delta delta))
+            {
+                throw new Exception($"Delta not found '{cid}'");
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4585091)